client/src/App.jsx

```diff
@@ -157,23 +162,69 @@ function App() {
     fetchGameReviews();
   }, [selectedGameId, newReviewSubmitted]);
 
-  // Fetch all user data based on logged-in user id
-  useEffect(() => {
-    const fetchAllUserData = async () => {
-      if (isLoggedIn && loggedInUser && favoritedGames.length === 0) {
-        try {
-          const response = await axios.get(
-            `${baseURL}/user-info/${loggedInUser.user_id}`
-          );
-          setFullLoggedInUserData(response.data);
-          await getFavoritedGames(loggedInUser.user_id);
-        } catch (error) {
-          console.error("Error fetching user data:", error);
-        }
-      }
-    };
+  // Extract fetchAllUserData into its own function
+const fetchAllUserData = async () => {
+  if (isLoggedIn && loggedInUser) {
+    try {
+      const response = await axios.get(
+        `${baseURL}/user-info/${loggedInUser.user_id}`
+      );
+      setFullLoggedInUserData(response.data);
+      await getFavoritedGames(loggedInUser.user_id);
+    } catch (error) {
+      console.error("Error fetching user data:", error);
+    }
+  }
+};
+
+// Call fetchAllUserData initially in useEffect to load data when necessary
+useEffect(() => {
+  if (isLoggedIn && loggedInUser && favoritedGames.length === 0) {
     fetchAllUserData();
-  }, [isLoggedIn, loggedInUser, baseURL, favoritedGames]);
+  }
+}, [isLoggedIn, loggedInUser]);
+
+   // Delete a user's review
+   const deleteUserReview = async (reviewId) => {
```

Comment:
I noticed that there is a lot of management/logic happening in your App.jsx, such as deleting a user's review.  In future implementation, moving the logic into a component allows the app to focus on higher-level functionality, like routing and state management, while the component handles the game and reviews logic.